### DISCUSSION TITLE
How can I use CUDA with NodeJS bindings?

### DISCUSSION THREAD

**Discussion by fijipants** [microsoft/onnxruntime#11067] [EP Q&A]
[It seems like onnxruntime-node should support the 'cuda' execution provider.](https://github.com/microsoft/onnxruntime/blob/49b329e26676e4a8f831e6829d4f893f78169f71/js/common/lib/inference-session.ts#L164)

Yet when I try to use cuda as an execution provider:

```ts
return await ort.InferenceSession.create(uri, {
    executionProviders: ['cuda']
});
```

It can't load them.

```
C:\repo\onnxruntime\js\common\dist\webpack:\onnxruntime-common\lib\backend-impl.ts:93
  throw new Error(`no available backend found. ERR: ${errors.map(e => `[${e.name}] ${e.err}`).join(', ')}`);
        ^
Error: no available backend found. ERR:
```

After looking through the code, I don't even see support for cuda, only for the other three: [cpu](https://github.com/microsoft/onnxruntime/blob/e9c68d57ca9978a7016fbec2e94931d13a6d2238/js/node/lib/index.ts#L8), [webgl](https://github.com/microsoft/onnxruntime/blob/e9c68d57ca9978a7016fbec2e94931d13a6d2238/js/web/lib/index.ts#L14), [wasm](https://github.com/microsoft/onnxruntime/blob/e9c68d57ca9978a7016fbec2e94931d13a6d2238/js/web/lib/index.ts#L18).

Is there any possible way I can use CUDA with onnxruntime through NodeJS?

**Comment by carpemonf** [comment]
Same issue here. @fijipants did you find any solution?
